src/ansys/mapdl/core/cli/hpc.py

```diff
@@ -0,0 +1,340 @@
+    exclusive: bool = None,
+    max_execution_time: int = None,
+    wait: bool = False,
+    debug: bool = False,
+    mode: Optional[Union["python", "shell", "apdl"]] = None,
+    output_to_json: Optional[bool] = False,
+):
+    import json
+
+    from ansys.mapdl.core.hpc.pyhps import (
+        PyMAPDLJobSubmission,
+        get_value_from_json_or_default,
+    )
+
+    if debug:
+        logging.basicConfig(
+            format="[%(asctime)s | %(levelname)s] %(message)s", level=logging.DEBUG
+        )
+
+    if config_file is None:
+        config_file = os.path.join(os.getcwd(), "hps_config.json")
+        logger.debug(f"Using default HPS configuration file: {config_file}")
+
+    url = get_value_from_json_or_default(url, config_file, "url", None)
+    user = get_value_from_json_or_default(user, config_file, "user", None)
+    password = get_value_from_json_or_default(password, config_file, "password", None)
+    python = get_value_from_json_or_default(python, config_file, "python", 3)
+    name = get_value_from_json_or_default(name, config_file, "name", "My PyMAPDL job")
+
+    num_cores = get_value_from_json_or_default(num_cores, config_file, "num_cores", 1)
+    memory = get_value_from_json_or_default(memory, config_file, "memory", 100)
+    disk_space = get_value_from_json_or_default(
+        disk_space, config_file, "disk_space", 100
+    )
+    exclusive = get_value_from_json_or_default(
+        exclusive, config_file, "exclusive", False
+    )
+    max_execution_time = get_value_from_json_or_default(
+        max_execution_time, config_file, "max_execution_time", 0
+    )
+
+    job = PyMAPDLJobSubmission(
+        url=url,
+        user=user,
+        password=password,
+        main_file=main_file,
+        mode=mode,
+        inputs=inputs,
+        outputs=outputs,
+        requirements_file=requirements_file,
+        shell_file=shell_file,
+        extra_files=extra_files,
+        output_files=output_files,
+        python=python,
+        num_cores=num_cores,
+        memory=memory,
+        disk_space=disk_space,
+        exclusive=exclusive,
+        max_execution_time=max_execution_time,
+        name=name,
+    )
+
+    job.submit()
+
+    if save_config_file:
+        config = {
+            "url": url,
+            "user": user,
+            "password": password,
+            "python": python,
+            "name": name,
+            "num_cores": num_cores,
+            "memory": memory,
+            "disk_space": disk_space,
+            "exclusive": exclusive,
+            "max_execution_time": max_execution_time,
+        }
+
+        logger.debug(
+            f"Saving the following configuration to the config file ({config_file}):\n{config}"
+        )
+        with open(config_file, "w") as fid:
+            json.dump(config, fid)
+
+    proj = job.project
+    if not output_to_json:
+        print(
+            f"You can check your project by visiting: {url}/projects#/projects/{proj.id}/jobs"
```

Comment:
Messages should conclude with punctuation, but I don't enforce this when a colon is used to provide a URL or filename